### PR TITLE
Move the knative-sandbox org to CODEOWNERS

### PR DESCRIPTION
**Pull request by mattmoor**
/hold
cc @rhuss @lionelvillard

/assign @chizhg @chaodaiG

### CODE REVIEW

File: prow/knative/plugins.yaml
Comment (mattmoor) [inline prow/knative/plugins.yaml:19]: @chaodaiG @chizhg Please advise on whether I should drop this?

File: prow/knative/plugins.yaml
Comment (mattmoor) [inline prow/knative/plugins.yaml:29]: I am assuming `repos` is correct mostly based on the commented lines below.  @chizhg @chaodaiG any sanity checking would be wonderful 🙏

Comment (chaodaiG) [reply prow/knative/plugins.yaml:29]: I believe it's correct

Comment (mattmoor) [reply prow/knative/plugins.yaml:29]: There are a few other places that seem to use `repos:` this way, so I figured it might be, but other places have `orgs:` too 😅

File: prow/knative/plugins.yaml
Comment (chaodaiG) [reply prow/knative/plugins.yaml:19]: Not sure about the motivation. It reacts to `/lgtm`(When lgtm as approve is enabled) and `/approve` comments and approves PR. Keeping this on shouldn't hurt imo

**Review by chaodaiG — APPROVED** [review]
Based on my knowledge, this is good to go

File: prow/knative/plugins.yaml
Comment (mattmoor) [reply prow/knative/plugins.yaml:19]: Well, without OWNERS files it leaves a confusing "NOT APPROVED" message file before the PR is merged anyhow. 😅

Comment (mattmoor) [reply prow/knative/plugins.yaml:19]: See: https://github.com/knative-sandbox/net-contour/pull/434#issuecomment-782245261

File: prow/knative/plugins.yaml
Comment (chaodaiG) [reply prow/knative/plugins.yaml:29]: You are definitely right about this. Generally prow configs are pretty consistent among each other, but when it comes to each individual plugins, there is no such good cohesion, as each plugin gets to decide the structure for itself, as well as how to interpret them (https://github.com/kubernetes/test-infra/blob/master/prow/plugins/config.go). This imo had become a tech debt and cleaning up is not an easy task.

File: prow/knative/plugins.yaml
Comment (chaodaiG) [reply prow/knative/plugins.yaml:19]: ah. I see 😅

Comment (chaodaiG) [reply prow/knative/plugins.yaml:19]: If there are still repos don't have OWNERS file, maybe should drop it for now?

Comment (mattmoor) [reply prow/knative/plugins.yaml:19]: The goal would be to drop OWNERS from all repos, right now the CODEOWNERS files coexist, but with this PR the source of truth will cut over to CODEOWNERS as the source of truth and OWNERS will become meaningless.

**Comment by mattmoor** [comment]
/unhold

We can drop the approve plugin in a follow up

**Pull request merged**
